### DISCUSSION TITLE
Getting OSSL error when calling login API

### DISCUSSION THREAD

**Discussion by arthurolmos** [gladly-team/next-firebase-auth#431] [Q&A]
Hello guys!

I'm getting this error when I call the login API when rendering the page:
{"library":"DECODER routines","reason":"unsupported","code":"ERR_OSSL_UNSUPPORTED"}}

It's happening when setAuthCookies is called.

Anyone knows how to solve it?

**Comment by arthurolmos** [comment]
Ok, i found out that the problem was with the firebase sdk key that was not reading correctly.

Just to make clear for any struggling, in your .env.local, your firebase key must be in double quotes: '"-----BEGIN PRIVATE KEY-----..."'

And on init(): 
 privateKey: process.env.FIREBASE_PRIVATE_KEY
          ? JSON.parse(process.env.FIREBASE_PRIVATE_KEY)
          : undefined,

**Reply by kmjennison** [reply]
@arthurolmos Yes, that should work.
